src/aihwkit/inference/noise/pcm.py

```diff
@@ -102,6 +108,34 @@ def __init__(
         self.prog_noise_scale = prog_noise_scale
         self.read_noise_scale = read_noise_scale
         self.drift_scale = drift_scale
+        self.valid_kwargs = ['custom_drift_model']
+
+        if not all(key in self.valid_kwargs for key in kwargs):
+            ValueError("PCMLikeNoiseModel only supports kwargs = %s" % self.valid_kwargs)
+
+        self.custom_drift_model = kwargs.get('custom_drift_model')
```

Comment:
These "hidden" args are not very user friendly. Why not simply derive a new class from the PCMLikeNoiseModel that has a custom drift model? Then one does not get into problems with the checkpoint read of earlier files etc.